File: ArchitectureComponentsMVVM/app/src/main/java/com/andr/mvvm/views/UserListAdapter.java
package com.andr.mvvm.views;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.andr.mvvm.R;
import com.andr.mvvm.repository.User;

import java.util.List;

public class UserListAdapter extends RecyclerView.Adapter<UserListAdapter.UserViewHolder> {


    class UserViewHolder extends RecyclerView.ViewHolder {
        private final TextView userNameView;
        private  final TextView userOccupationView;

        private UserViewHolder(View itemView) {
            super(itemView);
            userNameView = itemView.findViewById(R.id.user_name);
            userOccupationView = itemView.findViewById(R.id.user_occ);
        }
    }

    private final LayoutInflater mInflater;
    private List<User> mUsers; // Cached copy of users


    UserListAdapter(Context context) { mInflater = LayoutInflater.from(context); }

    @NonNull
    @Override
    public UserViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View itemView = mInflater.inflate(R.layout.adapter_user_item, parent, false);
        return new UserViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull UserViewHolder holder, int position) {
        if (mUsers != null) {
            User current = mUsers.get(position);
            holder.userNameView.setText(current.getUserName());
            holder.userOccupationView.setText(current.getUserOccupation());
        } else {
            // Covers the case of data not being ready yet.
            holder.userNameView.setText("No Word");
        }
    }
    void setWords(List<User> users){
        mUsers = users;
        notifyDataSetChanged();
    }

    @Override
    public int getItemCount() {
        if (mUsers != null) {
            return mUsers.size();
        }else{
            return  0;
        }
    }
}

File: MVVM_Kotlin/app/src/main/java/com/mvvm/kot/Adapter/UserListAdapter.kt
package com.mvvm.kot.Adapter

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.mvvm.kot.R
import com.mvvm.kot.ViewModels.User
import kotlinx.android.synthetic.main.adapter_user_item.view.*
import kotlinx.android.synthetic.main.fragment_user_info.view.*
import kotlinx.android.synthetic.main.fragment_user_list.view.*

 public class UserListAdapter(): RecyclerView.Adapter<UserListAdapter.ViewHolder>() {

     var items:List<User>?=null
    override fun onCreateViewHolder(parent: ViewGroup?, viewType: Int): ViewHolder {
        return ViewHolder(LayoutInflater.from(parent?.context).inflate(R.layout.adapter_user_item, parent, false))
    }


    override fun onBindViewHolder(holder:ViewHolder?, position: Int) {
        holder?.userName_holder?.text = items!!.get(position).userName
        holder?.userOccupation_holder?.text = items!!.get(position).userOccupation
    }

    override fun getItemCount(): Int {
        if(items!=null) {
            return items!!.size
        }else{
            return 0
        }
    }

     public fun setListItems(items:List<User>?){
         this.items = items
         notifyDataSetChanged()
     }
    class ViewHolder (view: View) : RecyclerView.ViewHolder(view) {
        // Holds the TextView that will add each animal to
      val userName_holder = view.findViewById(R.id.user_name_item) as TextView
      val userOccupation_holder = view.findViewById(R.id.user_occ) as TextView
    }
}